src/procstat.c

```diff
@@ -736,12 +737,29 @@ void procstat_u64_series_add_point(struct procstat_series_u64 *series, uint64_t
 	int64_t delta;
 	int64_t delta2;
 	int64_t avg_delta;
-	unsigned reset;
+	unsigned reset, reset_interval_expired;
+	uint64_t reset_interval;
+
+	reset_interval = __atomic_load_n(&series->reset_interval, __ATOMIC_RELAXED);
+
+	struct timespec cur_time;
+	if (clock_gettime( CLOCK_REALTIME, &cur_time) == -1) {
+		return -1;
```

Comment:
maybe i'm missing something here, but `procstat_u64_series_add_point()` returns `void`, so how do you return an `int` here? did it compiled for you?
btw, assuming negative value implies an error, do we really want to fail adding the point, just because we failed to calculate the last reset time?

Reply:
@alexs-lb reset_interval_expired is not actually requires please consder revising to:
if ((reset_interval > 0)&&(time_since_last_reset > reset_interval)) {
reset = 1
series->last_reset_time = cur_time.tv_sec;
} else {
reset = __atomic_load_n(&series->reset, __ATOMIC_RELAXED);

}

then you keep asking only on reset.

Reply:
@adir-lb  
Hmm.. I indeed didn't notice that and it compiled successfully. So the better implementation is that if gettime() returned error, reset_period is not checked, agree?

@oren-lb some words are missing from your comment. I guess that you mean that I should automic_load reset only if reset_interval_expired == 0. Correct?

Reply:
@alexs-lb sounds fine to me.